recipes-bsp/u-boot/u-boot-distro-boot/uEnv.txt.in

```diff
@@ -109,6 +112,108 @@ bootcmd_boot=if test ${bootscript_debug} != 1; then \
+                           run temp; \
+                           if test $? != "0"; then \
+                               env set fuse_status failed; \
+                           fi; \
+                       fi ; \
+                       setexpr index $index + 1; \
+                       if test "${index}" = "a"; then \
+                           env set index "10"; \
+                       fi; \
+                   done; \
+               elif test "${fuse_status}" = "need-reboot"; then \
+                   @@STATUS_CMD@@; \
+                   if test $? != "0"; then \
+                       env set fuse_status failed; \
+                   fi; \
+                   if test "${fuse_prog_close}" = "1" && test "${fuse_status}" != "failed"; then \
+                       if test "${fuse_dry_run}" = "1"; then \
+                           echo "DRY RUN: Closing is set"; \
+                       else \
+                           @@CLOSE_CMD@@; \
+                       fi; \
+                       if test $? != "0"; then \
+                           env set fuse_status failed; \
+                       fi; \
+                   fi; \
+               fi; \
+               env set save_vars 1; \
+               if test "${fuse_status}" = "pending"; then \
+                   env set fuse_status need-reboot; \
+               elif test "${fuse_status}" = "need-reboot"; then \
+                   env set fuse_status success; \
+               fi; \
+           fi || true
+
+read_fuses=if test -n "${fuse_num}"; then \
+               for num in ${fuse_num}; do \
+                   if env exists fuse_val_${num}; then \
+                       env set temp fuse cmp \\$fuse_bank_word_$num 0x\\$fuse_val_$num; \
+                       run temp; \
+                   else \
+                       false; \
+                   fi; \
+                   if test $? != "0"; then \
+                       env set temp fuse readm \\$fuse_bank_word_$num ${loadaddr}; \
+                       run temp; \
+                       setexpr.l fuse_val_${num} "*${loadaddr}"; \
+                       env set save_vars 1; \
+                   fi; \
+               done; \
+               tdx_is_closed; \
+               if test $? != "0"; then \
+                   if test "${fuse_val_close}" != "0"; then \
+                       env set save_vars 1; \
+                   fi; \
+                   env set fuse_val_close 0; \
+               else \
+                   if test "${fuse_val_close}" != "1"; then \
+                       env set save_vars 1; \
+                   fi; \
+                   env set fuse_val_close 1; \
+               fi; \
+           fi || true
+
+set_vars=if test "${save_vars}" = "1"; then \
+             for var in ${vars_to_del}; do \
+                 env set temp saved_$var=\\$$var; \
+                 run temp; \
+                 env delete ${var}; \
+             done; \
+             env set save_vars 0; \
```

Comment:
Maybe `env delete save_vars` would be better here?

Reply:
Sure, we can do that. I was thinking it might be beneficial in some way to retain this variable. But given how the logic is currently designed, there's not really any reason to do so. But, yeah let's just delete it.

Reply:
Done in V4.